icu4c/source/common/uloc.cpp

```diff
@@ -1655,12 +1655,39 @@ uloc_getParent(const char*    localeID,
                char* parent,
                int32_t parentCapacity,
                UErrorCode* err)
+{
+    if (U_FAILURE(*err)) {
+        return 0;
+    }
+
+    CheckedArrayByteSink sink(parent, parentCapacity);
+    ulocimp_getParent(localeID, sink, err);
+
+    int32_t reslen = sink.NumberOfBytesAppended();
+
+    if (U_FAILURE(*err)) {
+        return reslen;
+    }
+
+    if (sink.Overflowed()) {
+        *err = U_BUFFER_OVERFLOW_ERROR;
+    } else {
+        u_terminateChars(parent, parentCapacity, reslen, err);
+    }
+
+    return reslen;
```

Comment:
FYI/later: We should also have a function that takes a sink+output buffer and does all of this stuff.